dashboard/lib/service/appengine_cocoon.dart

```diff
@@ -146,23 +147,36 @@ class AppEngineCocoonService implements CocoonService {
   }
 
   @override
-  Future<bool> rerunTask(Task task, String idToken) async {
-    assert(idToken != null);
+  Future<CocoonResponse<bool>> rerunTask(Task task, String idToken) async {
+    if (idToken == null || idToken.isEmpty) {
+      return const CocoonResponse<bool>.error('Need to sign in to perform rerun operation');
+    }
 
     final QualifiedTask qualifiedTask = QualifiedTask.fromTask(task);
     assert(qualifiedTask.isLuci);
 
-    /// This endpoint only returns a status code.
+    //this endpoint returns CocoonResponse<bool> which reflects whether the api
+    //call returns success, returns bad status code, or throws an exception
     final Uri postResetTaskUrl = apiEndpoint('/api/reset-prod-task');
-    final http.Response response = await _client.post(postResetTaskUrl,
-        headers: <String, String>{
-          'X-Flutter-IdToken': idToken,
-        },
-        body: jsonEncode(<String, String>{
-          'Key': task.key.child.name,
-        }));
-
-    return response.statusCode == HttpStatus.ok;
+    try {
+      final http.Response response = await _client.post(postResetTaskUrl,
+          headers: <String, String>{
+            'X-Flutter-IdToken': idToken,
+          },
+          body: jsonEncode(<String, String>{
+            'Key': task.key.child.name,
+          }));
+      final bool success = response.statusCode == HttpStatus.ok;
+      return success
+          ? const CocoonResponse<bool>.data(true)
+          : CocoonResponse<bool>.error('bad status code is ' +
+              response.statusCode.toString() +
+              ', response body is ' +
+              response.body.toString() +
+              '\n');
+    } catch (error, stacktrace) {
```

Comment:
Catching errors should be specific (we should specify the type like `HttpException error`. https://pub.dev/documentation/http/latest/http/post.html indicates post will not return exceptions, so what are we trying to catch?

Reply:
got it. I think in 87661 discussion it was mentioned that we want to retrieve stack trace as well. I thought we would get stack trace if we encounter an exception. Umm maybe my understanding is off, so if exceptions are not thrown then we wouldn't get any stack trace?

Reply:
There's no underlying issues we're not catching. The issue is we don't surface this information (so moving this from `bool` to `CocoonResponse<bool` is enough. Additionally, we'll need to add a description of the issue if there is an error

Reply:
Thanks for explaining! So essentially we don't need exceptions, and merely describing the various kinds of bad http return code would be enough?